a3fe/run/simulation.py

```diff
@@ -131,6 +131,9 @@ def __init__(
                 f"Lambda value {self.lam} not in list of lambda values: {self.engine_config.lamvals}"
             )
 
+        self.timestep = self.engine_config.timestep
+        self.nrg_freq = self.engine_config.energy_frequency
```

Comment:
Try and remove all instances of self.timestep and self.nrg_freq, and replace them with self.engine_config.timestep etc - this means the information is stored only in one place and is easier to update.